include/exadg/convection_diffusion/user_interface/application_base.h

```diff
@@ -131,6 +131,12 @@ class ApplicationBase
     return grid;
   }
 
+  std::shared_ptr<Grid<dim>>
+  get_grid_non_const()
+  {
+    return grid;
+  }
```

Comment:
This is somewhat identical to making `grid` a public member of `ApplicationBase`. I am against such subtle changes, which  actually change the design a lot (since responsibilities become unclear).

We need to think about two options: (i) should the application execute the relevant functions changing the triangulation during AMR, or (ii) should the driver be responsible for the `Grid` object, meaning that `grid` becomes a member variable of `Driver` and is handed over to `ApplicationBase` e.g. to create the grid.

Reply:
this is still open! separate PR moving `grid` needed.

Reply:
`grid` was moved to the driver in #596.
this member function was removed